src/Native/Runtime/arm/PInvoke.S

```diff
@@ -22,17 +22,22 @@ NESTED_ENTRY RhpPInvoke, _TEXT, NoHandler
         str     lr, [r0, #OFFSETOF__PInvokeTransitionFrame__m_RIP]
         str     r7, [r0, #OFFSETOF__PInvokeTransitionFrame__m_FramePointer]
         str     sp, [r0, #OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs]
+        mov     r3, #PTFF_SAVE_SP
+        str     r3, [r0, #OFFSETOF__PInvokeTransitionFrame__m_dwFlags]
 
         PROLOG_PUSH "{r5,lr}"
 
-        mov   r5, r0
+        mov     r5, r0
         // get TLS global variable address
+        // r0 = GetThread()
         INLINE_GETTHREAD
         str     r0, [r5, #OFFSETOF__PInvokeTransitionFrame__m_pThread]
         str     r5, [r0, #OFFSETOF__Thread__m_pTransitionFrame]
+
         ldr     r3, =C_FUNC(RhpTrapThreads)
         ldr     r3, [r3]
-        cbnz    r3, LOCAL_LABEL(InvokeRareTrapThread)
+        cmp     r3, #TrapThreadsFlags_TrapThreads
```

Comment:
The test against `TrapThreadsFlags_TrapThreads` would have to be bit test, not comparison.

I think it is better to keep this as comparison with `TrapThreadsFlags_None`. It should be find keep it as cbnz and just mention `TrapThreadsFlags_None` in a comment.